### PR TITLE
[PHP 8.4] Add mb_* functions

### PR DESCRIPTION
**Pull request by saundefined**

### CODE REVIEW

File: language-snippets.ent [hunk 1]
```diff
@@ -3774,6 +3774,47 @@ local: {
 '>
 
 <!-- strings snippets -->
+<!ENTITY strings.stripped.characters '
+ <itemizedlist xmlns="http://docbook.org/ns/docbook">
+    <listitem>
+     <simpara>
+      " " (<acronym>ASCII</acronym> <literal>32</literal>
```
Comment (Girgias) [inline language-snippets.ent:3781]: Literal tag? (ditto for every other entry)

File: reference/mbstring/functions/mb-lcfirst.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ </refsect1>
+
+ <refsect1 role="seealso">
+  &reftitle.seealso;
+  <para>
+   <simplelist>
+    <member><function>mb_ucfirst</function></member>
+    <member><function>lcfirst</function></member>
+   </simplelist>
+  </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-lcfirst.xml:53]: Useless para tag
```suggestion
  <simplelist>
   <member><function>mb_ucfirst</function></member>
   <member><function>lcfirst</function></member>
  </simplelist>
```

File: reference/mbstring/functions/mb-lcfirst.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    </varlistentry>
+   </variablelist>
+  </para>
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the resulting string.
+  </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-lcfirst.xml:43]: New style guide guidelines
```suggestion
  <simpara>
   Returns the resulting string.
  </simpara>
```

File: reference/mbstring/functions/mb-lcfirst.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    <varlistentry>
+     <term><parameter>string</parameter></term>
+     <listitem>
+      <para>
+       The input string.
+      </para>
+     </listitem>
+    </varlistentry>
+   </variablelist>
+  </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-lcfirst.xml:36]: Useless para tag

File: reference/mbstring/functions/mb-lcfirst.xml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+   <type>string</type><methodname>mb_lcfirst</methodname>
+   <methodparam><type>string</type><parameter>string</parameter></methodparam>
+  </methodsynopsis>
+  <para>
+   Performs a multi-byte safe <function>lcfirst</function> operation,
+   and returns a string with the first character of
+   <parameter>string</parameter> lowercased if that character is
+   an ASCII character in the range <literal>"A"</literal> (0x41) to
+   <literal>"Z"</literal> (0x5a).
+  </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-lcfirst.xml:20]:
```suggestion
  <simpara>
   Performs a multi-byte safe <function>lcfirst</function> operation,
   and returns a string with the first character of
   <parameter>string</parameter> lowercased if that character is
   an ASCII character in the range <literal>"A"</literal> (0x41) to
   <literal>"Z"</literal> (0x5a).
  </simpara>
```

**Review by Girgias — CHANGES_REQUESTED** [review]
Markup changes that need to be replicated to all new files.

**Comment by youkidearitai** [comment]
Thank you very much for adding mb_* functions.

Note: `mb_trim`, `mb_ltrim` and `mb_rtrim` functions signature are changed from RFC <https://wiki.php.net/rfc/mb_trim> to https://github.com/php/php-src/pull/13820

File: reference/mbstring/functions/mb-ltrim.xml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+   </varlistentry>
+   <varlistentry>
+    <term><parameter>characters</parameter></term>
+    <listitem>
+     <para>
+      You can also specify the characters you want to strip, by means of the
+      <parameter>characters</parameter> parameter.
+      Simply list all characters that you want to be stripped. With
+      <literal>..</literal> you can specify a range of characters.
+     </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-ltrim.xml:46]: The documentation should not use personalization (i.e. "you"). And considering this is repeated, might as well be an entity.

File: reference/mbstring/functions/mb-lcfirst.xml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+ <refsect1 role="parameters">
+  &reftitle.parameters;
+  <variablelist>
+   <varlistentry>
+    <term><parameter>string</parameter></term>
+    <listitem>
+     <para>
+      The input string.
+     </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-lcfirst.xml:32]:
```suggestion
     <simpara>
      The input string.
     </simpara>
```

File: reference/mbstring/functions/mb-lcfirst.xml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><parameter>encoding</parameter></term>
+    <listitem>
+     <para>
+      The string encoding.
+     </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-lcfirst.xml:40]:
```suggestion
     <simpara>
      The string encoding.
     </simpara>
```

File: language-snippets.ent [hunk 9]
```diff
@@ -3774,6 +3774,41 @@ local: {
 '>
 
 <!-- strings snippets -->
+<!ENTITY strings.stripped.characters '
+ <itemizedlist xmlns="http://docbook.org/ns/docbook">
+    <listitem>
+     <simpara>
+      " " (<acronym>ASCII</acronym> 32 (0x20)), an ordinary space.
```
Comment (Girgias) [inline language-snippets.ent:3781]: Add literal tags, same for the other cases.
```suggestion
      <literal>" "</literal> (<acronym>ASCII</acronym> 32 (0x20)), an ordinary space.
```

**Review by Girgias — CHANGES_REQUESTED** [review]
More remarks and markup nits

File: language-snippets.ent [hunk 10]
```diff
@@ -3774,6 +3774,50 @@ local: {
...
+   </itemizedlist>
+'>
+
+<!ENTITY strings.parameter.characters.optional '
+ <simpara xmlns="http://docbook.org/ns/docbook">
+      Optionally, the stripped characters can also be specified using
+      the <parameter>characters</parameter> parameter.
+      Simply list all characters that need to be stripped. With
+      <literal>..</literal> it is possible to specify an incrementing range of characters.
+ </simpara>
```
Comment (Girgias) [inline language-snippets.ent:3818]:
```suggestion
 <simpara xmlns="http://docbook.org/ns/docbook">
  Optionally, the stripped characters can also be specified using
  the <parameter>characters</parameter> parameter.
  Simply list all characters that need to be stripped.
  With <literal>..</literal> it is possible to specify an incrementing range of characters.
 </simpara>
```

File: reference/mbstring/functions/mb-rtrim.xml [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+ <refsect1 role="parameters">
+  &reftitle.parameters;
+  <variablelist>
+   <varlistentry>
+    <term><parameter>string</parameter></term>
+    <listitem>
+     <para>
+      The input string.
+     </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-rtrim.xml:36]:
```suggestion
     <simpara>
      The input string.
     </simpara>
```

File: reference/mbstring/functions/mb-trim.xml [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+ <refsect1 role="parameters">
+  &reftitle.parameters;
+  <variablelist>
+   <varlistentry>
+    <term><parameter>string</parameter></term>
+    <listitem>
+     <para>
+      The <type>string</type> that will be trimmed.
+     </para>
```
Comment (Girgias) [inline reference/mbstring/functions/mb-trim.xml:34]:
```suggestion
     <simpara>
      The <type>string</type> that will be trimmed.
     </simpara>
```

**Review by Girgias — APPROVED** [review]
Final markup review, but LGTM otherwise.

**Comment by saundefined** [comment]
Thanks a lot!

**Pull request merged**